samples/subsys/llext/edk/app/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: Apache-2.0
+
+list(APPEND LLEXT_EDK_REMOVE_FLAGS -mcpu=cortex-r5 -mfloat-abi=hard)
```

Comment:
doesn't sound right to me to remove float abi flags.
What will happen later when loading the llext library ?

This seems like a recipe for a ABI incompatibility disaster.

Reply:
The sample actually works without those, but not with them - maybe this is a bug somewhere else? qemu? toolchain? llext loader? I've added a comment about why it's needed on the file.

Reply:
> The sample actually works without those, but not with them 

How do you know ?

Have you ensured that your test sample actually uses any floating point instruction or is it just a dummy sample which happens to work because it doesn't use floatingpoint at all ?

So far i'm not concerned that we are just pushing some code which happens to work in a small testcase, but which may actually cause some hard to debug issues later.

I need a bit more explanation of why this is needed or a finer restriction for platforms that breaks.
So far this is not sufficient imo:
```
# A permission fault happens on qemu_cortex_r5 when using these
# flags. Remove them for now while we figure out the issue.
```

Reply:
So I did a bit more investigation around this, and it seems that it's related to some relocation instructions not currently supported by llext - and removing those flags happen to cause the compiler to not generate code using those instructions, hence why the sample does work without them.  Created a new issue for that at #72832 - and noted that at the comment for the hack.

Note that this workaround is applied only for the sample - it's not done for the EDK itself - and should be revisited when issue is fixed.

Reply:
> Note that this workaround is applied only for the sample - it's not done for the EDK itself - and should be revisited when issue is fixed.

I know, but it's the only example in-tree we will have, so users will look at this and use it as reference code.
Thus a bit of extra care must be taken.

Thanks for the additional input, comment, and reference to the opened issue 👍